src/st/cat/e_breakable.c

```diff
@@ -1,4 +1,147 @@
+    EXPLOSION_SMALL,          EXPLOSION_SMALL,
+    EXPLOSION_SMALL,          EXPLOSION_SMALL_MULTIPLE,
+    EXPLOSION_SMALL_MULTIPLE, EXPLOSION_SMALL_MULTIPLE,
+    EXPLOSION_SMALL_MULTIPLE, EXPLOSION_SMALL_MULTIPLE,
+    EXPLOSION_SMALL_MULTIPLE, EXPLOSION_SMALL_MULTIPLE};
+static u16 palettes[] = {
+    PAL_NONE,      PAL_NONE,      PAL_NONE,      PAL_BREAKABLE, PAL_BREAKABLE,
+    PAL_BREAKABLE, PAL_BREAKABLE, PAL_BREAKABLE, PAL_BREAKABLE, PAL_BREAKABLE};
+static u16 anim_sets[] = {
+    ANIMSET_OVL(2),  ANIMSET_OVL(2),  ANIMSET_OVL(2),  ANIMSET_OVL(12),
+    ANIMSET_OVL(12), ANIMSET_OVL(12), ANIMSET_OVL(12), ANIMSET_OVL(12),
+    ANIMSET_OVL(12), ANIMSET_OVL(12), ANIMSET_OVL(12), ANIMSET_OVL(2)};
+static u16 unk_5A[] = {0, 0, 0, 91, 91, 91, 91, 91, 91, 91};
+static u8 draw_modes[] = {
+    DRAW_TPAGE | DRAW_TPAGE2, DRAW_TPAGE | DRAW_TPAGE2,
+    DRAW_TPAGE | DRAW_TPAGE2, DRAW_TPAGE | DRAW_TPAGE2,
+    DRAW_TPAGE | DRAW_TPAGE2, DRAW_TPAGE | DRAW_TPAGE2,
+    DRAW_TPAGE | DRAW_TPAGE2, DRAW_TPAGE | DRAW_TPAGE2,
+    DRAW_TPAGE | DRAW_TPAGE2, DRAW_TPAGE | DRAW_TPAGE2,
+    DRAW_TPAGE | DRAW_TPAGE2, DRAW_DEFAULT};
+static u16 z_priorities[] = {168, 168, 96, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+static u16 hitbox_offsets_y[12] = {0};
+
+extern EInit OVL_EXPORT(EInitBreakable);
+
+void OVL_EXPORT(EntityBreakable)(Entity* self) {
+    Entity* entity;
+    s16* debrisOffsetsY;
+    u16 breakableType;
+    s32 debrisIndex;
+    s32 debrisCount;
+    s16 posY;
+
+    breakableType = self->params >> 12;
+    if (!self->step) {
+        InitializeEntity(OVL_EXPORT(EInitBreakable));
+        self->zPriority = g_unkGraphicsStruct.g_zEntityCenter - 20;
+        if (z_priorities[breakableType]) {
+            self->zPriority = z_priorities[breakableType];
+        }
+        self->drawMode = draw_modes[breakableType];
+        self->hitboxHeight = hitbox_heights[breakableType];
+        self->animSet = anim_sets[breakableType];
+        self->unk5A = unk_5A[breakableType];
+        self->palette = palettes[breakableType];
+#ifdef INVERTED_STAGE
+        self->hitboxOffY = -hitbox_offsets_y[breakableType];
+#else
+        self->hitboxOffY = hitbox_offsets_y[breakableType];
+#endif
+    }
+    AnimateEntity(animations[breakableType], self);
+    if (self->hitParams) {
+        entity = AllocEntity(&g_Entities[224], &g_Entities[256]);
+        if (entity != NULL) {
+            CreateEntityFromCurrentEntity(E_EXPLOSION, entity);
+            entity->params = explosion_types[breakableType];
+        }
+        switch (breakableType) {
+        case BUST:
+            g_api.PlaySfx(SFX_GLASS_BREAK_E);
+            entity = AllocEntity(&g_Entities[160], &g_Entities[192]);
+            if (entity != NULL) {
+                CreateEntityFromEntity(E_HEART_DROP, self, entity);
+                entity->params = self->params & 0x1FF;
+            }
+            PreventEntityFromRespawning(self);
+            DestroyEntity(self);
+            return;
```

Comment:
These seems bizarre enough to warrant a comment. As written, the sfx will never play.

Reply:
I believe this may have been a copy paste mistake on my part.  I'll verify it and update it.